src/lbaf/Model/lbsBlock.py

```diff
@@ -0,0 +1,76 @@
+import sys
+
+from ..Utils.exception_handler import exc_handler
+
+
+class Block:
+    """ A class representing a memory block with footprint and home
+    """
+    def __init__(
+        self,
+        i: int,
+        h_id: int,
+        size: float=0.0,
+        o_ids: set=set()):
+
+        # Block index
+        if not isinstance(i, int) or isinstance(i, bool):
+            sys.excepthook = exc_handler
+            raise TypeError(
+                f"i: incorrect type {type(i)}")
+        else:
+            self.__index = i
+
+        # Rank to which block is initially assigned
+        if not isinstance(h_id, int) or isinstance(i, bool):
```

Comment:
Are there missing parentheses here? Should this be `h_id` instead of `i` on the right?

Reply:
Good catch, should be `h_id` indeed!